File: LeetCode-Problems/Easy/Decode-XOR-Array.py
# Python
class Solution(object):
    def decode(self, encoded, first):
        """
        :type encoded: List[int]
        :type first: int
        :rtype: List[int]
        """
        decodedArray = list()
        decodedArray.append(first)
        temp = 0;
        for x in range (0, len(encoded)):
            temp = encoded[x] ^ decodedArray[x]
            decodedArray.append(abs(temp))
        return decodedArray

File: LeetCode-Problems/Easy/Decode-XOR-Array.java
//Java
class Solution {
    public int[] decode(int[] encoded, int first) {
        int[] decodedArray = new int[encoded.length+1];
        decodedArray[0] = first;
        int temp = 0;
        
        for(int i = 0; i < encoded.length; i++){
            temp = encoded[i] ^ decodedArray[i];
            decodedArray[i+1] = Math.abs(temp);
        }
        return decodedArray;
    }
}
